Anima/diffusion/mcm_estimator/animaMCMEstimator.cxx

```diff
@@ -184,17 +194,11 @@ int main(int argc,  char **argv)
     filter->SetGTolerance(gTolArg.getValue());
     filter->SetMaxEval(maxEvalArg.getValue());
 
-    filter->SetUseConcentrationBoundsFromDTI(false);
-    filter->SetUseFixedWeights(fixWeightsArg.isSet() && (mlModeArg.getValue() != 2));
-
-    filter->SetFreeWaterProportionFixedValue(freeWaterWeightArg.getValue());
-    filter->SetStationaryWaterProportionFixedValue(stationaryWaterWeightArg.getValue());
-    filter->SetRestrictedWaterProportionFixedValue(restrictedWaterWeightArg.getValue());
-    filter->SetStaniszProportionFixedValue(staniszWeightArg.getValue());
-
     filter->SetUseConstrainedDiffusivity(fixDiffArg.isSet());
     filter->SetUseConstrainedFreeWaterDiffusivity(!optFWDiffArg.isSet());
     filter->SetUseConstrainedIRWDiffusivity(!optIRWDiffArg.isSet());
+    filter->SetUseConstrainedStaniszDiffusivity(!optStaniszDiffArg.isSet());
```

Comment:
I would remove all the options `SetUseConstrained*` and use the diffusivity setters for that purpose. For example If a diffusivity is set to -1 (or any negative number, which could be the default), then it is estimated, otherwise it is constrained to the input value.

Reply:
Unless you think of the setter for getting an initial value only (as I might understand from above).

Reply:
The setters are here for initial values only, and the constraints have their own parameters. If you set the initial value to something and fix the parameters, then it will keep its value. I prefer this option rather than having fake negative diffusivities.